ui/src/app/views/sign-in-flow-template/component.tsx

```diff
@@ -0,0 +1,152 @@
+  },
+  'accountCreation': {
+    backgroundImgSrc: '/assets/images/create-account-male.png',
+    smallerBackgroundImgSrc: '/assets/images/create-account-male-standing.png'
+  },
+  'accountCreationSuccess': {
+    backgroundImgSrc: '/assets/images/congrats-female.png',
+    smallerBackgroundImgSrc: 'assets/images/congrats-female-standing.png'
+  }};
+
+const headerImg = '/assets/images/logo-registration-non-signed-in.svg';
+
+export const RegistrationPageTemplateReact = withWindowSize()(
+  class extends React.Component<PageTemplateProps, PageTemplateState> {
+
+    constructor(props: PageTemplateProps) {
+      super(props);
+      this.state = {
+        currentStep: 'login',
+        invitationKey: '',
+        profile: {} as Profile
+      };
+      this.setCurrentStep = this.setCurrentStep.bind(this);
+      this.onKeyVerified = this.onKeyVerified.bind(this);
+      this.setProfile = this.setProfile.bind(this);
+    }
+
+    componentDidMount() {
+      document.body.style.backgroundColor = '#e2e3e5';
+      this.props.onInit();
+    }
+
+    nextDirective(index) {
+      switch (index) {
+        case 'login':
+          return <LoginReactComponent signIn={this.props.signIn} onCreateAccount={() =>
+                                     this.setCurrentStep('invitationKey')}/>;
+        case 'invitationKey':
+          return <InvitationKey onInvitationKeyVerify={(key) => this.onKeyVerified(key)}/>;
+        case 'accountCreation':
+          return <AccountCreation invitationKey={this.state.invitationKey}
+                                  setProfile={this.setProfile}/>;
+        case 'accountCreationSuccess':
+          return <AccountCreationSuccess profile={this.state.profile}/>;
+        default:
+          return;
+        }
+      }
+
+    setCurrentStep(nextStep) {
+      this.setState({
+        currentStep: nextStep
+      });
+    }
+
+    onKeyVerified(invitationKey) {
+      this.setState({
+        invitationKey: invitationKey,
+        currentStep: 'accountCreation'
+      });
+    }
+
+    setProfile(profile) {
+      this.setState({
+        profile: profile,
+        currentStep: 'accountCreationSuccess'
+      });
+    }
+
+    render() {
+      return <div style={styles.signedInContainer}>
+        <div style={{width: '100%', display: 'flex', flexDirection: 'column'}}>
+          <div style={styles.template(this.props.windowSize, pageImages[this.state.currentStep])}>
+            <img style={{height: '1.75rem', marginLeft: '1rem', marginTop: '1rem'}}
+                 src={headerImg}/>
+            <div style={{flex: '0 0 41.66667%', maxWidth: '41.66667%', minWidth: '25rem'}}>
+              {this.nextDirective(this.state.currentStep)}
+            </div>
+          </div>
+        </div>
+      </div>;
+    }
+});
+
+export default RegistrationPageTemplateReact;
+
+@Component({
+  template: '<div #root></div>'
+})
+export class SignInTemplateComponent extends ReactWrapperBase {
```

Comment:
I agree with the template part but trying to figure out if it still should be SignInComponent. This is because signing in is just a part of the whole flow, this is also an entry point for registration.